x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go

```diff
@@ -80,6 +87,14 @@ datasources:
         - paths: /var/log/mysql/access.log
           dataset: dsds
           index: mytype-dsds-nsns
+  - name: All specified with empty strings
+    namespace: ""
```

Comment:
Shouldn't the expected now be `default`?

Reply:
this is test for injecting index, it does not change existing values. 
we either generate index based on these values or we're adding `add_fields` processor with correct values

Reply:
we need both. values and index must match. 

In a follow up we can discuss, if we should even error if it is "".

Reply:
@michalpristas I also thought the result should be default?

Reply:
i think we can fix it but outside of injectIndex rule, what this should do is injecting index and not modifying anything on top of that.